pkg/k8sutils/templates/pod/spec.go

```diff
@@ -265,6 +265,7 @@ func Spec(spec v1.SpecInterface, container corev1.Container, volumes []corev1.Vo
 		ServiceAccountName:            spec.GetServiceAccount(),
 		TerminationGracePeriodSeconds: spec.GetTerminationGracePeriodSeconds(),
 		Affinity:                      spec.GetAffinity(),
+		TopologySpreadConstraints:     spec.GetTopologySpreadConstraints(),
```

Comment:
Since this is a newer feature on k8s. If the user is using an older version of k8s, such as version 1.18, will it cause deployment to fail?

Reply:
I have verified it.
```
# deployments.apps "kube-starrocks-operator" was not valid:
# * patch: Invalid value: "map[spec:map[aaaa:[]]]": strict decoding error: unknown field "spec.aaaa"```

Reply:
TopologySpreadContraints was promoted to beta in Kubernetes 1.18 as far as I can find so quickly. Which was released 4 years ago, or almost half the life of Kubernetes :). I think most projects have given up on supporting such old versions of Kubernetes since it can prevent projects from moving forwards with features. It might be possible to have this feature only get enabled after doing a Kubernetes version check, but I'm not familiar with the code enough to do that quickly, nor am I sure it's worth the effort for maintaining support for Kubernetes versions that people shouldn't have been using anymore for years now.

Reply:
https://kubernetes.io/blog/2020/05/introducing-podtopologyspread/
From this blog, topologySpreadConstraints was introduced in 1.18, which is the oldest version starrocks operator support.